src/pages/index.tsx

```diff
@@ -1,12 +1,29 @@
+import { GetServerSideProps } from 'next';
+import App from '../App';
+import { parseEmbedQuery } from '../utils';
+
 /**
  * This is a shim for now. Our app currently doesn't use
  * next routing - we're just loading the entire app in _app.tsx
  *
  * This will be fixed in a later PR, adding routes with server
  * rendering to grab OG images.
  */
-const HomePage = () => {
-  return null;
+const HomePage = ({ query }: { query: any }) => {
+  const embed = parseEmbedQuery(query);
+  return (
+    <App
+      embed={{
+        ...embed,
+        isEmbedded: false,
+        embedUrl: null,
+      }}
+    />
+  );
+};
+
+export const getServerSideProps: GetServerSideProps = async (ctx) => {
+  return { props: { query: ctx.query } };
 };
```

Comment:
note: at the moment we don't do "true" SSR so this doesn't quite have to include `getServerSideProps` (the query will be available client-side anyway). By removing that we'd be able to optimize this because this index route could be statically generated - whereas right now it has to be server-side rendered.

OTOH if we ever start to truly SSR this then this will become required.

Reply:
this needs to be read on the server-side because otherwise, the embedded mode will first load normally and then `useRouter` will pick up available query params and UI will have a flaky jump between the 2 modes.